src/bestcrypt_ve_fmt_plug.c

```diff
@@ -0,0 +1,456 @@
+
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+	for (index = 0; index < count; index++) {
+		unsigned char kdf_out[32];
+		/* BestCrypt uses CBC mode with a null IV */
+		unsigned char iv[16] = {0};
+		unsigned char out[0x60];
+		unsigned char intermediate_keys[64];
+
+		SHA256_CTX ctx;
+		unsigned char sha256_hash[32];
+
+#ifdef _OPENMP
+		int t = omp_get_thread_num();
+		if (t >= max_threads) {
+			failed = -1;
+			continue;
+		}
+#else
+		const int t = 0;
+#endif
+		yescrypt_kdf(NULL, &local[t],
+		             (const uint8_t *)saved_key[index],
+		             strlen(saved_key[index]),
+		             (const uint8_t *)cur_salt->salt,
+		             0x18,
+		             &params,
+		             kdf_out, 0x20);
+
+		/*
+		   we will now use output of scrypt as key for desired encryption
+		   algorithm in CBC mode
+		*/
+		if (cur_salt->enc_algoID == aesId) {
+			AES_KEY aes_key;
+
+			AES_set_decrypt_key(kdf_out, 256, &aes_key);
+			AES_cbc_encrypt(cur_salt->encrypted_data,
+			                out,
+			                0x60,
+			                &aes_key,
+			                iv,
+			                AES_DECRYPT);
+		} else if(cur_salt->enc_algoID == twofishId) {
+			Twofish_key tkey;
+
+			Twofish_prepare_key(kdf_out, 32, &tkey);
+			Twofish_Decrypt_no_padding(&tkey,
+			                           cur_salt->encrypted_data,
+			                           out,
+			                           0x60,
+			                           iv);
+		} else if(cur_salt->enc_algoID == serpentId) {
+			uint8_t ks[SERPENT_KS];
+
+			serpent_set_key(kdf_out, ks);
+			serpent_cbc_decrypt(cur_salt->encrypted_data,
+			                    out,
+			                    0x60,
+			                    ks,
+			                    iv);
+		} else if(cur_salt->enc_algoID == camelliaId) {
+			CAMELLIA_KEY ck;
+
+			Camellia_set_key(kdf_out, 256, &ck);
+			Camellia_cbc_encrypt(cur_salt->encrypted_data,
+			                     out,
+			                     0x60,
+			                     &ck,
+			                     iv,
+			                     CAMELLIA_DECRYPT);
+		} /* else if(cur_salt->enc_algoID == rc6Id) {
+			TODO
+		}
+		*/
+
+		/* we now compute sha256(decrypted_content[0:0x40]) and 
+		compare it with decrypted_content[0x40:0x60] */
+		memcpy(intermediate_keys, out, 0x40);
+		SHA256_Init(&ctx);
+		SHA256_Update(&ctx, intermediate_keys, 0x40);
+		SHA256_Final(sha256_hash, &ctx);
+		cracked[index] = (0 == memcmp(sha256_hash, out + 0x40, 0x20));
+	}
+
+	if (failed) {
+#ifdef _OPENMP
+		if (failed < 0) {
+			fprintf(stderr, "OpenMP thread number out of range\n");
+			error();
+		}
+#endif
+	}
+
+	return count;
+}
+
+static int cmp_all(void *binary, int count)
+{
+	int index;
+
+	for (index = 0; index < count; index++)
+		if (cracked[index])
+			return 1;
+	return 0;
+}
+
+static int cmp_one(void *binary, int index)
+{
+	return cracked[index];
+}
+
+static int cmp_exact(char *source, int index)
+{
+	return 1;
+}
```

Comment:
BTW, I think we should introduce a `default_cmp_exact` like this and use it in all formats that have it dummy like this. Please feel free to make a separate PR with that.